test/fixtures/browser-app/app.js

```diff
@@ -7,4 +7,6 @@ var loopback = require('loopback');
 var boot = require('../../../');
 
 var app = module.exports = loopback();
-boot(app);
+app.start = function(done) {
+  boot(app, __dirname, done);
+};
```

Comment:
When the `app` is used as a dependency or library, it will be much more flexible to defer the boot process. It's very similar as the factory pattern. I would prefer to minimize the side effect of `require` so that we can do things such as recycling the app during test. If the `app` is the main program, we can use `if(module === require.main)` to run the boot automatically.